tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/grpc/sample/ProtoBuilderTestCase.java

```diff
@@ -0,0 +1,134 @@
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.test.service.grpc.sample;
+
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.test.BaseTest;
+import org.ballerinalang.test.util.TestUtils;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Test class for proto file builder compiler plugin.
+ */
+@Test(groups = "grpc-test")
+public class ProtoBuilderTestCase extends BaseTest {
+
+    @BeforeClass
+    private void setup() throws Exception {
+        TestUtils.prepareBalo(this);
+    }
+
+    @Test(description = "Test compiler plugin for unary service with primitive params.")
+    public void testUnaryServiceWithPrimitiveParams() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "unary_server1.bal");
+        CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
+        Assert.assertEquals(result.getErrorCount(), 0, "Compilation errors in unary_server1.bal");
+        Assert.assertEquals(result.getAST().getServices().size(), 1, "File should have one service defined.");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().size(), 1, "Service " +
+                "node should have one default annotation");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().get(0).getAnnotationName
+                ().getValue(), "ServiceDescriptor");
+    }
+
+    @Test(description = "Test compiler plugin for unary service with header params.")
+    public void testUnaryServiceWithHeaderParams() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "unary_service3.bal");
+        CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
+        Assert.assertEquals(result.getErrorCount(), 0, "Compilation errors in unary_server3.bal");
+        Assert.assertEquals(result.getAST().getServices().size(), 1, "File should have one service defined.");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().size(), 1, "Service " +
+                "node should have one default annotation");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().get(0).getAnnotationName
+                ().getValue(), "ServiceDescriptor");
+    }
+
+    @Test(description = "Test compiler plugin for server streaming service.")
+    public void testServerStreamingService() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "server_streaming_service" +
+                ".bal");
```

Comment:
Lets not break at middle of the string value, rather break the line at one of the Paths argument string level.